Release workflow fails with "nothing to commit, working tree clean" when re-releasing an existing version

**Describe the bug**

When the release workflow is triggered with a version that has already been tagged, the \"Check tag does not exist\" step is supposed to abort the run gracefully. Instead, the prior step — \"Commit and push version bump\" — fails first with exit code 1 because `plugin.json` already contains the target version (written by the original release), leaving the working tree clean and nothing to commit.

**To Reproduce**

1. Identify a version that has already been released and tagged (e.g. `0.1.0`).
2. Go to **Actions → Release → Run workflow** and enter that version.
3. Observe the \"Commit and push version bump\" step fail.

**Expected behavior**

The \"Check tag does not exist\" step should run **before** any write operations (version bump commit/push). If the tag exists it should abort with:

```
Tag v<VERSION> already exists. Aborting.
```

**Screenshots**

<!-- Screenshot to be attached -->

**Additional context**

Root cause: the \"Check tag does not exist\" step (`.github/workflows/release.yml` line 58) is ordered after \"Commit and push version bump\". The guard never gets a chance to fire when the version is already released.

Error output from the failing run:

```
Run git config user.name "github-actions[bot]"
  git config user.name "github-actions[bot]"
  git config user.email "github-actions[bot]@users.noreply.github.com"
  git add plugins/token-effort/.claude-plugin/plugin.json
  git commit -m "chore: release v$VERSION"
  git push origin HEAD
  shell: /usr/bin/bash -e {0}
  env:
    GH_TOKEN: ***
    VERSION: 0.1.0
On branch main
Your branch is up to date with 'origin/main'.

nothing to commit, working tree clean
Error: Process completed with exit code 1.
```

--- a/.github/workflows/release.yml
+++ b/.github/workflows/release.yml
@@ -41,6 +41,13 @@ jobs:
           token: ${{ steps.release-manager-token.outputs.token }}
           fetch-depth: 1
 
+      - name: 🔍 Check tag does not exist
+        run: |
+          if git ls-remote --tags origin "refs/tags/v$VERSION" | grep -q .; then
+            echo "Tag v$VERSION already exists. Aborting."
+            exit 1
+          fi
+
       - name: 📝 Patch plugin.json version
         run: |
           jq --arg version "$VERSION" '.version = $version' \
@@ -52,15 +59,8 @@ jobs:
           git config user.name "github-actions[bot]"
           git config user.email "github-actions[bot]@users.noreply.github.com"
           git add plugins/token-effort/.claude-plugin/plugin.json
-          git commit -m "chore: release v$VERSION"
+          git diff --cached --quiet || git commit -m "chore: release v$VERSION"
           git push origin HEAD
-
-      - name: 🔍 Check tag does not exist
-        run: |
-          if git ls-remote --tags origin "refs/tags/v$VERSION" | grep -q .; then
-            echo "Tag v$VERSION already exists. Aborting."
-            exit 1
-          fi
 
       - name: 🏷️ Create git tag
         run: |

